### DISCUSSION TITLE
Search by criteria error vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/DefaultQuoteStrategy.php line 44

### DISCUSSION THREAD

**Discussion by nyatMeat** [doctrine/doctrine2#8803] [Support Questions]
I have an issue
vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/DefaultQuoteStrategy.php line 44

```php

    /**
     * @param $teamId
     * @return Team|null
     */
    public function getTeamByAccount(Account $account): ?Team
    {
        $criteria = Criteria::create()->where(Criteria::expr()->eq("account", $teamId));

        $collection = $this->getTeams()->matching($criteria);
        if ($collection->isEmpty()) {
            return null;
        }

        return $collection->first();
    }
```

The code of the Team entity

```php
/**
 * @ORM\Entity(repositoryClass=TeamRepository::class)
 * @ORM\Table(name="team")
 * @ORM\HasLifecycleCallbacks()
 */
class Team
{
    use TimestampableTrait;

    /**
     * @var \Ramsey\Uuid\UuidInterface
     *
     * @ORM\Id
     * @ORM\Column(type="uuid", unique=true)
     * @ORM\GeneratedValue(strategy="CUSTOM")
     * @ORM\CustomIdGenerator(class=UuidGenerator::class)
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $name;

    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     */
    private $description;


    /**
     * @ORM\ManyToOne(targetEntity=Account::class)
     * @ORM\JoinColumn(nullable=false)
     */
    private $account;
    
    /**
     * @ORM\ManyToOne(targetEntity=User::class)
     * @ORM\JoinColumn(nullable=false)
     */
    private $manager;
```
The code of the Account entity
```php
/**
 * @ORM\Entity(repositoryClass=AccountRepository::class)
 * @ORM\Table(name="account")
 * @ORM\HasLifecycleCallbacks()
 */
class Account
{

    use TimestampableTrait;

    /**
     * @var \Ramsey\Uuid\UuidInterface
     *
     * @ORM\Id
     * @ORM\Column(type="uuid", unique=true)
     * @ORM\GeneratedValue(strategy="CUSTOM")
     * @ORM\CustomIdGenerator(class=UuidGenerator::class)
     */
    private $id;

    /**
     * @ORM\Column(name="name", type="string", length=255)
     */
    private $name;


    /**
     * @ORM\Column(name="state", type="string", length=255)
     */
    private $state;
```

And the code of the User entity
```php
class User
{


    use TimestampableTrait;

    /**
     * @var \Ramsey\Uuid\UuidInterface
     *
     * @ORM\Id
     * @ORM\Column(type="uuid", unique=true)
     * @ORM\GeneratedValue(strategy="CUSTOM")
     * @ORM\CustomIdGenerator(class=UuidGenerator::class)
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=180, unique=true)
     */
    private $email;

    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     */
    private $firstName = '';

    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     */
    private $lastName = '';

    /**
     * @ORM\ManyToMany(targetEntity=Account::class)
     */
    private $accounts;

    /**
     * @ORM\ManyToMany(targetEntity=Team::class)
     */
    private $teams;
```
What could be wrong? As I read in articles about using criteria in the Entity everything should be fine, but it doesn't

Could someone help me to understand what the wrong here?

**Comment by greg0ire** [comment]
> I have an issue
vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/DefaultQuoteStrategy.php line 44

Uuuuh… that's not an issue, that's a filename.

> Search by criteria error

I doubt this is the actual error message you are getting.

If by chance you are using Symfony, consider reading this: https://symfony.com/doc/current/contributing/code/stack_trace.html , it might help you help us help you.

**Comment by nyatMeat** [comment]
> > I have an issue
> > vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/DefaultQuoteStrategy.php line 44
> 
> Uuuuh… that's not an issue, that's a filename.
> 
> > Search by criteria error
> 
> I doubt this is the actual error message you are getting.
> 
> If by chance you are using Symfony, consider reading this: https://symfony.com/doc/current/contributing/code/stack_trace.html , it might help you help us help you.

I'm sorry. Didn't copy the whole output.
The problem is 
```
Undefined index 'account'
```
But in entity Team `account` field is exists, and when I try to filter collection by criteria I am getting this error.
Might be something wrong with relations? I didn't find the answer on stackowerflow and github

**Reply by greg0ire** [reply]
Ok so maybe we lack a proper exception message here:

https://github.com/doctrine/orm/blob/1f6bfe175494ed88fb662c86ebd319ef66901ab8/lib/Doctrine/ORM/Mapping/DefaultQuoteStrategy.php#L41

It would be great to have something along the lines of `Unknown field "account" for entity X
It looks like you are assuming that X is `Team` here. Can you please try tweaking `DefaultQuoteStrategy` as follows?

```php
public function getColumnName($fieldName, ClassMetadata $class, AbstractPlatform $platform)
{
    if (!isset($class->fieldNames[$fieldName])) {
        throw new \UnexpectedValueException(sprintf(
            'Unknown field "%s" for entity "%s"',
            $fieldName,
            $class->name
        ));
    }
    …
```
It should help you figure out what X actually is.
If you are using Symfony, running `bin/console doctrine:schema:validate` is a good idea too.
